pkg/ldap/client.go

```diff
@@ -148,25 +148,58 @@ func (c *Client) LdapGet(ctx context.Context,
 	return entries[0], nil
 }
 
+func (c *Client) LdapGetWithStringDN(ctx context.Context,
+	searchDN string,
+	filter string,
+	attrNames []string,
+) (*ldap.Entry, error) {
+	entries, _, err := c.LdapSearchWithStringDN(ctx, ldap.ScopeBaseObject, searchDN, filter, attrNames, "", 1)
+	if err != nil {
+		return nil, err
+	}
+	if len(entries) == 0 {
+		return nil, fmt.Errorf("entry not found: %s", searchDN)
```

Comment:
Maybe these should be grpc statuses. eg: https://github.com/ConductorOne/baton-ldap/blob/main/pkg/ldap/client.go#L218

That way baton-sdk will retry or warn in some cases.